eng/pipelines/templates/steps/archetype-sdk-docs.yml

```diff
@@ -0,0 +1,151 @@
+
+          # Create Various Directoies for Doc Generation
+          Write-Host "Create Directories Required for Doc Generation"
+          mkdir "$($SafeName)/dll-docs/my-api"
+          mkdir "$($SafeName)/dll-docs/dependencies/my-api"
+          mkdir "$($SafeName)/dll-xml-output"
+          mkdir "$($SafeName)/dll-yaml-output"
+          mkdir "$($SafeName)/docfx-output"
+
+          # Build Packages for Doc Generation
+          if ('${{ parameters.LibType }}' -eq '') { # Client Package
+            Write-Host "Build Packages for Doc Generation - Client"
+            dotnet build "$(Build.SourcesDirectory)/eng/service.proj" /p:ServiceDirectory=$PackageLocation /p:IncludeTests=false /p:IncludeSamples=false /p:OutputPath="$(Build.BinariesDirectory)/$($SafeName)/dll-docs/my-api" /p:TargetFramework=netstandard2.0
+
+            # Include Dependencies
+            dotnet build "$(Build.SourcesDirectory)/eng/service.proj" /p:ServiceDirectory=$PackageLocation /p:IncludeTests=false /p:IncludeSamples=false /p:OutputPath="$(Build.BinariesDirectory)/$($SafeName)/dll-docs/dependencies/my-api" /p:TargetFramework=netstandard2.0 /p:CopyLocalLockFileAssemblies=true
+          }
+
+          if ('${{ parameters.LibType }}' -eq 'Management') { # Management Package
+            Write-Host "Build Packages for Doc Generation - Management"
+            dotnet msbuild "eng/mgmt.proj" /p:scope=$PackageLocation /p:OutputPath="$(Build.BinariesDirectory)/$($SafeName)/dll-docs/my-api" -maxcpucount:1 -nodeReuse:false
+
+            # Include Dependencies
+            dotnet msbuild "eng/mgmt.proj" /p:scope=$PackageLocation /p:OutputPath="$(Build.BinariesDirectory)/$($SafeName)/dll-docs/dependencies/my-api" /p:CopyLocalLockFileAssemblies=true -maxcpucount:1 -nodeReuse:false
+          }
+
+          # Remove all unneeded artifacts from build directory
+          Write-Host " Remove all unneeded artifacts from build output directory"
+          Remove-Item –Path "$(Build.BinariesDirectory)/$($SafeName)/dll-docs/my-api/*" -Include * -Exclude ${{artifact.name}}.dll, ${{artifact.name}}.xml
+
+          # Initialize Frameworks File
+          Write-Host "Initialize Frameworks File"
+          ./mdoc/mdoc.exe fx-bootstrap "$($SafeName)/dll-docs"
+
+          # Include XML Files
+          Write-Host "Include XML Files"
+          ./PopImport/popimport.exe -f "$($SafeName)/dll-docs"
+
+          # Produce ECMAXML
+          Write-Host "Produce ECMAXML"
+          ./mdoc/mdoc.exe update -fx "$($SafeName)/dll-docs" -o "$($SafeName)/dll-xml-output" --debug -lang docid -lang vb.net -lang fsharp --delete
+
+          # Produce Generate YAML
+          Write-Host "Generate YAML"
+          ./ECMA2Yml/ECMA2Yaml.exe -s "$($SafeName)/dll-xml-output" -o "$($SafeName)/dll-yaml-output"
+
+          # Provision DocFX Directory
+          Write-Host "Provision DocFX Directory"
+          ./docfx/docfx.exe init -q -o "$($SafeName)/docfx-output/docfx_project"
```

Comment:
One word of caution while running a number of executables in a PS script you will need to check the exit codes otherwise if there is a failure the rest of the script will attempt to continue and may fail in weird ways, or worse pass with incorrect results.